crates/context/config/src/client/env/proxy/types/starknet.rs

```diff
@@ -397,3 +446,146 @@ impl From<StarknetProposalId> for ProposalId {
+                    }
+                }
+
+                // Get value array length and contents
+                if let Some(value_len) = iter.next() {
+                    let value_len = value_len.to_bytes_be()[31] as usize;
+                    let mut value = Vec::new();
+                    for _ in 0..value_len {
+                        if let Some(felt) = iter.next() {
+                            value.push(*felt);
+                        }
+                    }
+                    entries.push((key, value));
+                }
+            }
+        }
+
+        Ok(Self { entries })
+    }
+}
+
+impl From<(Vec<Felt>, Vec<Felt>)> for ContextStorageEntry {
+    fn from((key_felts, value_felts): (Vec<Felt>, Vec<Felt>)) -> Self {
+        let key = key_felts
+            .iter()
+            .flat_map(|f| f.to_bytes_be())
+            .filter(|&b| b != 0)
+            .collect();
+
+        let value = value_felts
+            .iter()
+            .flat_map(|f| f.to_bytes_be())
+            .filter(|&b| b != 0)
+            .collect();
+
+        ContextStorageEntry { key, value }
+    }
+}
+
+struct ValueCodec<'a>(&'a serde_json::Value);
+
+impl<'a> ValueCodec<'a> {
+    fn encode_string<W: FeltWriter>(s: &str, writer: &mut W) -> Result<(), Error> {
+        if s.starts_with("0x") {
+            // Attempt to handle hex string directly as a single Felt
+            if let Ok(felt) = Felt::from_hex(s) {
+                writer.write(felt);
+                return Ok(());
+            }
+        }
+        // Regular string - split into chunks
+        let chunk_size = 31;
+        let bytes = s.as_bytes();
+
+        // Write number of chunks first
+        writer.write(Felt::from(bytes.len() / chunk_size));
+
+        // Write each chunk as a Felt
+        for chunk in bytes.chunks(chunk_size) {
+            writer.write(Felt::from_bytes_be_slice(chunk));
+        }
+        Ok(())
+    }
+}
+
+impl<'a> Encode for ValueCodec<'a> {
+    fn encode<W: FeltWriter>(&self, writer: &mut W) -> Result<(), Error> {
+        match self.0 {
+            serde_json::Value::String(s) => Self::encode_string(s, writer),
+            serde_json::Value::Object(obj) => {
+                writer.write(Felt::from(obj.len()));
+                for (key, value) in obj {
+                    Self::encode_string(key, writer)?;
+                    ValueCodec(value).encode(writer)?;
+                }
+                Ok(())
+            }
+            serde_json::Value::Bool(b) => {
+                writer.write(Felt::from(*b as u64));
+                Ok(())
+            }
+            serde_json::Value::Number(n) => {
+                if let Some(n) = n.as_u64() {
+                    writer.write(Felt::from(n));
+                } else {
+                    // For other numbers, convert to string and encode as string
+                    let s = n.to_string();
+                    ValueCodec(&serde_json::Value::String(s)).encode(writer)?;
```

Comment:
This will only fail when the number is negative or a float, how are those typically represented in Felt?

Is it first converted to string?